src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -93,100 +94,93 @@ public virtual void GetPolicies(JsonSerializerOptions options)
 
         private void DeterminePropertyName(JsonSerializerOptions options)
         {
-            if (PropertyInfo != null)
+            if (PropertyInfo == null)
             {
-                JsonPropertyNameAttribute nameAttribute = GetAttribute<JsonPropertyNameAttribute>();
-                if (nameAttribute != null)
-                {
-                    NameAsString = nameAttribute.Name;
+                return;
+            }
 
-                    // null is not valid; JsonClassInfo throws an InvalidOperationException after this return.
-                    if (NameAsString == null)
-                    {
-                        return;
-                    }
-                }
-                else if (options.PropertyNamingPolicy != null)
-                {
-                    NameAsString = options.PropertyNamingPolicy.ConvertName(PropertyInfo.Name);
+            JsonPropertyNameAttribute nameAttribute = GetAttribute<JsonPropertyNameAttribute>(PropertyInfo);
+            if (nameAttribute != null)
+            {
+                NameAsString = nameAttribute.Name;
 
-                    // null is not valid; JsonClassInfo throws an InvalidOperationException after this return.
-                    if (NameAsString == null)
-                    {
-                        return;
-                    }
-                }
-                else
+                // null is not valid; JsonClassInfo throws an InvalidOperationException after this return.
```

Comment:
It is possible. I'll change it. The reason why it was like this is because exceptions thrown here were wrapped by a `TargetInvocationException` since this method is called by the ctor which is called via reflection. I'll refactor the PropertyInfo classes to not have any logic in their ctor.